Add no_bots preset for crashsite

Soggs opened the pull request:
The preset removes access to construction and logistic bots on crashsite

Soggs committed:
Add no_bots preset for crashsite

The preset removes access to construction and logistic bots on crashsite

grilledham commented on map_gen/maps/crash_site/scenario_no_bots.lua:
```diff
@@ -0,0 +1,954 @@
+    game.surfaces.cutscene.set_tiles(tiles)
+end
+
+local function init(config)
```
config could have a flag for no_bot_outposts. Then we just don't add `mini_t1_robotics_factory` to `mini2` and `mini3` tables. The order in the tables doesn't matter, so it could be the last entry.

Is that the only different in this file compared to scenario.lua, would that allow us to avoid the duplication?

Soggs replied:
So had a moment to think and I remembered when going over it, that it might be an idea to generally move the list of outposts that get loaded into the config. So we are generally able to alter outposts based on config. Partially because it might interesting to have different levels of allowed bots. This current preset is literally no bots but maybe we only want to prohibit logistic bots.

Also would leave for presets like no science outposts or no military outposts.

grilledham replied:
Sounds like a good idea, more effort though. I'll leave it up to you if you want to do that.

grilledham replied:
I would prefer not to have a duplicate scenario file. I think adding a flag to config and not adding the bot outposts would be easy to do.

grilledham commented on map_gen/maps/crash_site/presets/no_bots.lua:
```diff
@@ -0,0 +1,53 @@
+local Event = require 'utils.event'
+
+local config = {
+    scenario_name = 'crashsite',
```
We should also add the preset to scenario_templates. Whatever you call that preset, probably "crashsite-no-bots", should be the name here.

grilledham commented on map_gen/maps/crash_site/presets/no_bots.lua:
```diff
@@ -0,0 +1,53 @@
+}
+
+local Scenario = require 'map_gen.maps.crash_site.scenario_no_bots'
+ScenarioInfo.set_map_name('Crashsite')
```
Change the name here too.

Soggs committed:
Added only personal bots preset for Crashsite

Added a preset that does only unlock personal roboports and construction robots.

Soggs committed:
Added the new presets to the scenario templates

Soggs committed:
Updated names inside the preset configs

Soggs commented:
I think I have everything updated. In the mean time I did add a second preset, that is just personal construction bots instead of no bots at all.

Soggs committed:
Merge branch 'master' into crashsite-no-bots

Soggs committed:
Updated presets to use the new dynamic outpost lists

- Updated no_bots preset
- Updated only_personal_construction preset
- Removed scenario_no_bots.lua as it became unnecessary.

Soggs committed:
Added only construction bots preset

Soggs committed:
Added logistic network embargo preset

Soggs committed:
Added Steam all the way preset

Soggs committed:
Enabled Kovarex for nuclear weapons and fuel.

Soggs committed:
Added Raining Bullets preset

Soggs committed:
Also disbled discharge defense research
